.svglintrc.js

```diff
@@ -163,6 +164,136 @@ module.exports = {
+                    if (upperShorthandCurveCommand === command && x1Coord === xPrevCoord && y1Coord === yPrevCoord && x1Coord === x2Coord && y1Coord === y2Coord) {
+                      return true;
+                    }
+                    // Absolute bézier curve (C) having the same coordinate as the previous segment and last control point equal to the ending point
+                    if (upperCurveCommand === command && x1Coord === xPrevCoord && y1Coord === yPrevCoord && x2Coord === xCoord && y2Coord === yCoord) {
+                      return true;
+                    }
+                  }
+
+                  return (
+                    // Absolute horizontal direction (H) having the same x coordinate as the previous segment
+                    (upperHorDirectionCommand === command && x1Coord === xPrevCoord) ||
+                    // Absolute vertical direction (V) having the same y coordinate as the previous segment
+                    (upperVerDirectionCommand === command && x1Coord === yPrevCoord) ||
+                    // Absolute movement (M or L) having the same coordinate as the previous segment
+                    (upperMovementCommands.includes(command) && x1Coord === xPrevCoord && y1Coord === yPrevCoord)
+                  );
+                }
+              }
+            };
+            const invalidSegments = segments.filter(getInvalidSegments);
+
+            if (invalidSegments.length) {
+              invalidSegments.forEach(([command, x1Coord, y1Coord, ...rest]) => {
+                let readableSegment = `${command}${x1Coord}`;
+                if (y1Coord !== undefined) {
+                  readableSegment += ` ${y1Coord}`;
+                }
+                if (curveCommands.includes(command)) {
+                  const [x2Coord, y2Coord, xCoord, yCoord] = rest;
+                  readableSegment += `, ${x2Coord} ${y2Coord}`;
+                  if (yCoord !== undefined) {
+                    readableSegment += `, ${xCoord} ${yCoord}`;
+                  }
+                  if (command === lowerShorthandCurveCommand && (x2Coord !== 0 || y2Coord !== 0)) {
+                    readableSegment += ` (should be "l${x2Coord} ${y2Coord}" or removed)`;
+                  }
+                  if (command === upperShorthandCurveCommand) {
+                    readableSegment += ` (should be "L${x2Coord} ${y2Coord}" or removed)`;
+                  }
+                  if (command === lowerCurveCommand && (xCoord !== 0 || yCoord !== 0)) {
+                    readableSegment += ` (should be "l${xCoord} ${yCoord}" or removed)`;
+                  }
+                  if (command === upperCurveCommand) {
+                    readableSegment += ` (should be "L${xCoord} ${yCoord}" or removed)`;
+                  }
+                }
+                reporter.error(`Unexpected segment ${readableSegment} in path.`);
```

Comment:
What about improve the error message wrapping all failed segments between double quotes and showing the resolution tip at the end?

So the message `Unexpected segment c0 0, 0 0, 0 0.007 (should be "l0 0.007" or removed) in path.` would be converted to `Unexpected segment "c0 0, 0 0, 0 0.007" in path (should be "l0 0.007" or removed).`.

```suggestion
                let readableSegment = `${command}${x1Coord}`,
                    resolutionTip = '';
                if (y1Coord !== undefined) {
                  readableSegment += ` ${y1Coord}`;
                }
                if (curveCommands.includes(command)) {
                  const [x2Coord, y2Coord, xCoord, yCoord] = rest;
                  readableSegment += `, ${x2Coord} ${y2Coord}`;
                  if (yCoord !== undefined) {
                    readableSegment += `, ${xCoord} ${yCoord}`;
                  }
                  if (command === lowerShorthandCurveCommand && (x2Coord !== 0 || y2Coord !== 0)) {
                    resolutionTip = `should be "l${x2Coord} ${y2Coord}" or removed`;
                  }
                  if (command === upperShorthandCurveCommand) {
                    resolutionTip = `should be "L${x2Coord} ${y2Coord}" or removed`;
                  }
                  if (command === lowerCurveCommand && (xCoord !== 0 || yCoord !== 0)) {
                    resolutionTip = `should be "l${xCoord} ${yCoord}" or removed`;
                  }
                  if (command === upperCurveCommand) {
                    resolutionTip = `should be "L${xCoord} ${yCoord}" or removed`;
                  }
                }
                let errorMessage = `Unexpected segment "${readableSegment}" in path`;
                if (resolutionTip) {
                  errorMessage += ` (${resolutionTip})`
                }
                errorMessage += '.';
                reporter.error(errorMessage);
```

Reply:
Much better, thanks for the suggestion!

Reply:
Sorry for the mix of variable names of three reviews. I'm seeing now meesages like `Ineffective segment "c0 0, 0 0, 0.007 -0.007should be "l.007 -.007" or removed" in path (should be removed).`. Seems that you should use `resolutionTip` variable in the conditionals as it's shown in this review.

Reply:
Oh, it is ok. I already fixed it, but did not pushed it 🤦‍♂️